static/js/components/SpinnerButton_test.js

```diff
@@ -17,32 +17,39 @@ describe("SpinnerButton", () => {
   })
 
   for (const promiseState of ["not_clicked", "rejected", "resolved"]) {
-    it(`passes through all props when promise state is ${promiseState}`, async () => {
-      const promise =
-        promiseState === "rejected" ? Promise.reject() : Promise.resolve()
-      const onClickPromise = () => promise
-      const props = {
-        "data-x":       "y",
-        onClickPromise: onClickPromise,
-        className:      "class1 class2"
-      }
-      const wrapper = shallow(
-        <SpinnerButton {...props}>childText</SpinnerButton>
-      )
-      const button = wrapper.find("button")
-      const buttonProps = button.props()
+    it(`passes through all props when promise state is ${promiseState}`, () => {
```

Comment:
Was the async function not working here?

Reply:
the rejected promise was causing an issue because we weren't handling the rejection, I rewrite it to return a `new Promise` just because it made it a little easier